clash-cores/src/Clash/Cores/SigmaDeltaADC.hs

```diff
@@ -0,0 +1,262 @@
+module Clash.Cores.SigmaDeltaADC (sigmaDeltaADC) where
+
+import Clash.Prelude hiding (filter, truncate)
+import Clash.Sized.Internal.BitVector
+
+-- | Shared functions
+-- Function to remove the LSBs from a BitVector
+truncate
+  :: (KnownNat n0, KnownNat n1)
+  => BitVector (n0 + n1)
+  -- ^ entire bitvector
+  -> BitVector n0
+  -- ^ bitvector without number of LSBs
+truncate input = output
+ where
+  (output, _) = split input
+
+-- | Remove the LSBs from the accumulator output when the accumulator is ready
+-- This is used for both the accumulator and the filter
+truncateAccum
+  :: (KnownNat n0, KnownNat n1)
+  => BitVector n0
+  -- ^ The previous accum
+  -> (BitVector (n0 + n1), Bool)
+  -- ^ parts of the tuple
+  --
+  -- 1. Accumulator output
+  -- 2. Boolean for when the accumulator is ready
+  -> (BitVector n0, BitVector n0)
+  -- ^ parts of tuple
+  --
+  -- 1. updated accum
+  -- 2. output accum
+truncateAccum accum (sigma, rollover)
+  | rollover = (sigma', accum)
+  | otherwise = (accum, accum)
+   where
+     sigma' = truncate sigma
+
+-- |Accumulator
+-- The main function for the accumulator,
+-- all functions that start with accumulator are used for the accumulator
```

Comment:
Tbh this documentation does not say anything about the accumulator itself.
I'd expect a short description of what it is and information that is relevant for someone to use this function.